### PR TITLE
[SPARK-2945][YARN][Doc]add doc for spark.executor.instances

### PR DESCRIPTION
**Pull request by WangTaoTheTonic**
https://issues.apache.org/jira/browse/SPARK-2945

spark.executor.instances works. As this JIRA recommended, we should add docs for this common config.

### CODE REVIEW

**Comment by WangTaoTheTonic** [comment]
@sryza Added docs for situation that dynamic allocation is enabled.

**Comment by WangTaoTheTonic** [comment]
retest this please.

**Comment by WangTaoTheTonic** [comment]
Seems like the UT has broken, but I think it is unralted with this patch.

File: docs/running-on-yarn.md [hunk 1]
```diff
@@ -105,6 +105,13 @@ Most of the configs are the same for Spark on YARN as for other deployment modes
   </td>
 </tr>
 <tr>
+ <td><code>spark.executor.instances</code></td>
+  <td>2</td>
+  <td>
+    The number of executors. Don't set this when dynamic allocation is enabled as they are not compatible.
```
Comment (andrewor14) [inline docs/running-on-yarn.md]: This is only used in YARN. I will add this when I merge

Comment (andrewor14) [inline docs/running-on-yarn.md]: oh wait this is the YARN page. LGTM

**Pull request closed without merging**